datafusion/common/src/scalar.rs

```diff
@@ -2695,14 +2730,22 @@ impl From<String> for ScalarValue {
     }
 }
 
-impl From<Vec<(&str, ScalarValue)>> for ScalarValue {
-    fn from(value: Vec<(&str, ScalarValue)>) -> Self {
-        let (fields, scalars): (SchemaBuilder, Vec<_>) = value
-            .into_iter()
-            .map(|(name, scalar)| (Field::new(name, scalar.data_type(), false), scalar))
-            .unzip();
+// Wrapper for ScalarValue::Struct that checks the length of the arrays, without nulls
```

Comment:
TODO: Remove these two.

Reply:
Are these still TODO?

Reply:
Yes, we haven't changed to Scalar<T> yet.